flybywiresim/a32nx
# fix(efb): landing calculator

Fixes #5723

## Summary of Changes

- Landing calculator:
  - Fixed the "Calculate" button's *disabled* state (make it not clickable when disabled).
  - Made all inputs initially empty (by having undefined values), so that the placeholders (/units) can be seen. It also avoids being able to press "Calculate" initially (*which avoids calculating with, previously, out of range values*).
  - Fixed the "Clear" button.
  - Fixed the "Wind Magnitude" input.
  - Improved the input placeholders.

## Screenshots (if necessary)



![EFB-empty](https://user-images.githubusercontent.com/15957088/144943822-a04d69ba-e12a-4b6b-aaaa-2db99626082c.png)

![EFB-filled](https://user-images.githubusercontent.com/15957088/144943832-b6e79382-a1d5-48e7-824e-8185a4956c3d.png)


## References





## Additional context


This PR contains the same fixes as PR #5740 , but without the toggle feature. It has a comparatively reduced scope and should therefore be easier to review. The mentioned PR will likely be closed in favor of this one and the upcoming FlypadOS v3.


Discord username (if different from GitHub): `bk67#6465`

## Testing instructions


1. Go to the landing calculator page in the EFB: all inputs should **initially be empty**, the "Calculate" button should be **disabled** (somewhat transparent and not clickable).
2. Input an ICAO in the "Airport ICAO" field (for ex: `LFST`).
3. Click "Get METAR".
4. "Wind Direction", "Wind Magnitude", "Temperature", "QNH" and "Weight" inputs should be filled in.
5. Fill in all "Rwy ..." inputs and the "Approach Speed" inputs (the missing ones).
6. The "Calculate" button should be enabled now (fully opaque).
7. Click "Calculate".
8. Make sure the distances seem consistent with the input values.
9. Change the wind direction to get a tailwind (relative to the runway heading), set the wind magnitude to 0. Click "Calculate". Set the wind magnitude to 10. Click "Calculate". **Make sure the landing distance had increased with the higher tailwind**.
10. Click "Clear".
11. Make sure all inputs are empty again (as initially).


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
